libr/socket/http_server.c

```diff
@@ -72,6 +74,23 @@ R_API RSocketHTTPRequest *r_socket_http_accept (RSocket *s, int accept_timeout,
 			} else
 			if (!strncmp (buf, "Content-Length: ", 16)) {
 				content_length = atoi (buf+16);
+			} else
+			if (!strncmp (buf, "Authorization: Basic ", 21) && httpauth) {
```

Comment:
use r_str_startswith()

Reply:
Fair point ... even though your solution need your suggestion implies calculating a known constant length (and I was following the general state of the rest of the workflow).